test/e2e/user_workload_monitoring_test.go

```diff
@@ -588,8 +588,29 @@ func assertTenancyForMetrics(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	for _, q := range []string{"up", `up{namespace="should-be-overwritten"}`, fmt.Sprintf(`up{namespace="%s"}`, userWorkloadTestNs)} {
-		t.Logf("Running query %q", q)
+	for _, tc := range []struct {
+		query     string
+		expStatus int
+	}{
+		{
+			// expect to add namespace label
```

Comment:
Instead of adding comments to describe each test, it would be better to add a name to the structure and expose it in the form of subtests via [t.Run()](https://pkg.go.dev/testing#T.Run).

Reply:
Thanks, good point.